apps/frontend/src/pages/Charities.tsx

```diff
@@ -0,0 +1,119 @@
+import CurrentCharity from "../components/CurrentCharity";
+
+export interface Organization {
+  name: string;
+  logoUrl: string;
+  donated: number;
+  receiving: boolean;
+}
+
+export interface Charity {
+  id: string;
+  name: string;
+  description: string;
+  startDate: string;
+  endDate: string;
+  imageUrl: string;
+  organizations: Organization[];
+  funds: number;
+  listingsCount: number;
+}
+
+const Charities = () => {
+  const currentDate = new Date();
+  const [showPastEvents, setShowPastEvents] = useState(false);
+  const [charities, setCharities] = useState<Charity[]>([]);
+  const [loading, setLoading] = useState(true);
+
+  useEffect(() => {
+    const fetchData = async () => {
+      try {
+        const response = await _axios_instance.get("/charities");
+        setCharities(response.data);
+        setLoading(false);
+      } catch (error) {
+        console.error("Error fetching charities:", error);
+        setLoading(false);
+      }
+    };
+    fetchData();
+  }, []);
+
+  const currentCharity = Array.isArray(charities)
+    ? charities.find((charity) => new Date(charity.endDate) >= currentDate)
+    : null;
+
+  const pastCharities = Array.isArray(charities)
+    ? charities.filter((charity) => new Date(charity.endDate) < currentDate)
+    : [];
+
+  const handleToggleChange = (event: ChangeEvent<HTMLInputElement>) => {
+    setShowPastEvents(event.target.checked);
+  };
+
+  if (loading) {
+    return <Spinner text="Loading charities..." />;
+  }
+
+  return (
+    <>
+      <SearchBar />
+      <FormControlLabel
+        control={
+          <Switch
+            checked={showPastEvents}
+            onChange={handleToggleChange}
+            color="primary"
+            id="charitiesSwitch"
+          />
+        }
+        label={
+          showPastEvents
+            ? "Show Current Charity Event"
+            : "Show Past Charity Events"
+        }
+        sx={{ marginLeft: "20px", marginTop: "20px" }}
+      />
+      {showPastEvents ? (
+        pastCharities.length > 0 ? (
+          <>
+            <Typography
+              variant="h5"
+              component="h3"
+              gutterBottom
+              sx={{ paddingLeft: "20px", paddingTop: "20px" }}
+            >
+              Past Charity Events
+            </Typography>
+            <Grid container spacing={2} justifyContent="center">
+              {pastCharities.map((charity, index) => (
+                <Grid item xs={12} sm={6} md={4} key={index}>
+                  <CharityCard {...charity} />
+                </Grid>
+              ))}
+            </Grid>
+          </>
+        ) : (
+          <Typography
+            variant="h6"
+            component="h4"
+            gutterBottom
+            sx={{ paddingLeft: "20px", paddingTop: "20px" }}
+          >
+            No Past Charities
+          </Typography>
+        )
+      ) : (
+        currentCharity && <CurrentCharity charity={currentCharity} />
```

Comment:
I think it'd be good to add this to the current charity toggle, that way the user can see the state at a glace and it's consistent with the other page
```suggestion
        <Box>
          <Typography
            variant="h5"
            component="h3"
            gutterBottom
            sx={{ paddingLeft: "20px", paddingTop: "20px" }}
          >
            Current Charity Events
          </Typography>
          {currentCharity && <CurrentCharity charity={currentCharity} />}
        </Box>
```
_You'll need to add the box import_